skeletons/classname.xml

```diff
@@ -1,105 +1,132 @@
- xmlns:xi="http://www.w3.org/2001/XInclude"
- xmlns="http://docbook.org/ns/docbook">
+<reference xml:id="class.classname" role="class" xmlns="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:xi="http://www.w3.org/2001/XInclude">
  <title>The ClassName class</title>
  <titleabbrev>ClassName</titleabbrev>
  
  <partintro>
- 
-<!-- {{{ ClassName intro -->
+
+  <!-- {{{ ClassName intro -->
   <section xml:id="classname.intro">
    &reftitle.intro;
-   <para>
+   <simpara>
     Description of the class.
-   </para>
+   </simpara>
   </section>
-<!-- }}} -->
- 
+  <!-- }}} -->
+
   <section xml:id="classname.synopsis">
    &reftitle.classsynopsis;
- 
-<!-- {{{ Synopsis -->
-   <classsynopsis>
-    <ooclass><classname>ClassName</classname></ooclass>
- 
-<!-- {{{ Class synopsis -->
-    <classsynopsisinfo>
-     <ooclass>
-      <classname>ClassName</classname>
-     </ooclass>
- 
-<!-- If the class extends another one, use this
-     <ooclass>
-      <modifier>extends</modifier>
-      <classname>BaseClassName</classname>
-     </ooclass>
--->
- 
-<!-- If the class implements an interface, or two, use this (one oointerface element per interface)
-     <oointerface>
-      <interfacename>InterfaceName</interfacename>
-     </oointerface>
--->
-    </classsynopsisinfo>
-<!-- }}} -->
- 
-    <classsynopsisinfo role="comment">Properties</classsynopsisinfo>
-<!-- If the property is documented below (xml:id=classname.props) use this
+
+   <!-- {{{ Synopsis -->
+   <classsynopsis class="class">
+    <ooclass>
+     <classname>ClassName</classname>
+    </ooclass>
+
+    <!-- If the class extends another one, use this -->
+    <ooclass>
+     <modifier>extends</modifier>
+     <classname>ParentClass</classname>
+    </ooclass>
+
+    <oointerface>
+     <modifier>implements</modifier>
+     <interfacename>Interface1</interfacename>
+    </oointerface>
+
+    <oointerface>
+     <interfacename>Interface2</interfacename>
+    </oointerface>
+
+    <oointerface>
+     <interfacename>Interface3</interfacename>
+    </oointerface>
+
+    <classsynopsisinfo role="comment">&Constants;</classsynopsisinfo>
     <fieldsynopsis>
      <modifier>public</modifier>
+     <modifier>const</modifier>
+     <type>int</type>
+     <varname linkend="classname.constants.some-constant">ClassName::SOME_CONSTANT</varname>
+    </fieldsynopsis>
+
+    <classsynopsisinfo role="comment">&Properties;</classsynopsisinfo>
+    <fieldsynopsis>
+     <modifier>protected</modifier>
      <type>string</type>
-     <varname linkend="classname.props.name">name</varname>
+     <varname linkend="classname.props.someproperty">SomeProperty</varname>
+     <initializer>""</initializer>
     </fieldsynopsis>
--->
- 
-<!-- If properties are documented "as a method", use this
-    <xi:include xpointer="xmlns(db=http://docbook.org/ns/docbook) xpointer(id('class.classname')/db:refentry/db:refsect1[@role='description']/descendant::db:fieldsynopsis)" />
--->
```

Comment:
Not sure about this removal. Properties probably have never been documented in separate files, though.

Reply:
mysqli does this AFAIK